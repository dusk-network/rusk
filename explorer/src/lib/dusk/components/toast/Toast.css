.dusk-toast {
  display: flex;
  flex-direction: column;
  position: fixed;
  right: 20px;
  bottom: 20px;
  z-index: 5;
  row-gap: 0.625em;
}

.dusk-toast__item {
  display: flex;
  column-gap: 0.625em;
  padding: 0.625em;
  border-radius: 1em;
  background-color: var(--background-color-alt);
  min-width: 200px;
  max-width: 250px;
}

.dusk-toast__item-icon-wrapper {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 1.625em;
  height: 1.625em;
  border-radius: 100%;
  flex-shrink: 0;
}

.dusk-toast__item-icon-wrapper--info {
  background-color: var(--info-color);
}

.dusk-toast__item-icon-wrapper--success {
  background-color: var(--success-color);
}

.dusk-toast__item-icon-wrapper--warning {
  background-color: var(--warning-color);
}

.dusk-toast__item-icon-wrapper--error {
  background-color: var(--error-color);
}

.dusk-toast__item-icon-wrapper--error .dusk-toast__item-icon {
  color: #fff;
}
