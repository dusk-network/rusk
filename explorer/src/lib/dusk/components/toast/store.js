import { get, writable } from "svelte/store";

/**
 * @type {import('svelte/store').Writable<ToastItem[]>}
 * @description Stores each toast as an object in the array
 */
export const toastList = writable([]);

/**
 * @type {import('svelte/store').Writable<Number>}
 * @description Stores the timer fused by all toasts
 */
export const toastTimer = writable(0);

/**
 * @param {string} message
 * @param {string} icon
 * @param {TooltipType} type
 */

function addToast(type, message, icon) {
  const id = `dusk-toast-${Math.random().toString(36)}`;

  toastList.update((store) => {
    return [
      ...store,
      {
        icon: icon,
        id: id,
        message: message,
        type: type,
      },
    ];
  });

  const timeoutID = window.setTimeout(() => {
    deleteToast(id, timeoutID);
  }, get(toastTimer));
}

/**
 * @param {string} id
 * @param {number} timeout
 */
function deleteToast(id, timeout) {
  window.clearTimeout(timeout);

  // Deletes toast from store queue
  toastList.update((store) => {
    return store.filter((toast) => toast.id !== id);
  });
}

export const toast = addToast;
