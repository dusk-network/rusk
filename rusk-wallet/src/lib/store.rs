// This Source Code Form is subject to the terms of the Mozilla Public
// License, v. 2.0. If a copy of the MPL was not distributed with this
// file, You can obtain one at http://mozilla.org/MPL/2.0/.
//
// Copyright (c) DUSK NETWORK. All rights reserved.

use std::io::Read;
use std::{fmt, fs};
use std::path::PathBuf;
use bytes::{BytesMut, BufMut, Bytes, Buf};

use dusk_pki::{SecretSpendKey, SecretKey};
use dusk_wallet_core::{Store, derive_sk, derive_ssk};

use crate::lib::errors::CliError;

/// Default directory name to store settings and keystore
const DATA_DIR: &str = ".dusk";

/// Stores all the user's settings and keystore in the file system
pub struct LocalStore {
    name: String,
    path: PathBuf,
    seed: [u8; 64],
}

impl Store for LocalStore {

    type Error = CliError;

    /// Retrieves the seed used to derive keys.
    fn get_seed(&self) -> Result<[u8; 64], Self::Error> {
        Ok(self.seed)
    }

    /// Retrieves a derived secret spend key from the store.
    fn retrieve_ssk(&self, index: u64) -> Result<SecretSpendKey, Self::Error> {
        Ok(derive_ssk(&self.seed, index))
    }

    /// Retrieves a derived secret key from the store.
    fn retrieve_sk(&self, index: u64) -> Result<SecretKey, Self::Error> {
        Ok(derive_sk(&self.seed, index))
    }

}

impl LocalStore {

    /// Creates a new store
    pub fn new(name: String, seed: [u8;64]) -> Result<LocalStore, CliError> {

        // construct path to file
        let home = dirs::home_dir().unwrap();
        let mut path = PathBuf::new();
        path.push(home.as_os_str());
        path.push(DATA_DIR);
        path.push(&name);
        path = path.with_extension("dat");

        // get the seed
        let mut seed_bytes= [0u8; 64];
        seed_bytes.copy_from_slice(&seed);

        // create the local store
        let store = LocalStore{
            name: name,
            path: path,
            seed: seed_bytes,
        };

        // save to disk and return
        store.save()?;
        Ok(store)

    }


    /// Loads wallet file from file
    pub fn from_file(path: PathBuf, _pwd: String) -> Result<LocalStore, CliError> {

        // basic sanity check
        if !path.is_file() {
            return Err(CliError::CorruptedFile);
        }

        // attempt to load and decode wallet
        let data = Bytes::from(fs::read(&path).unwrap());
        let mut reader = data.reader();

        // wallet_seed
        let mut wallet_seed = [0u8; 64];
        reader.read(&mut wallet_seed)?;

        // extract the name
        let name = path.file_name()
            .ok_or(CliError::CorruptedFile)?.to_str()
            .ok_or(CliError::CorruptedFile)?;

        // create and return
        Ok(LocalStore{
            name: String::from(name),
            path: path,
            seed: wallet_seed,
        })

    }

    /// Saves wallet to a file
    pub fn save(&self) -> Result<(), CliError> {

        let mut buf = BytesMut::new();

        // wallet seed (todo: encrypt)
        buf.put_slice(&self.seed);

        // write file
        fs::write(&self.path, &buf[..]).unwrap();
        Ok(())

    }

    /// The name of the wallet
    pub fn name(&self) -> String {
        self.name.clone()
    }

}

impl fmt::Debug for LocalStore {
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
        write!(f, "-------------------\n\
            LocalStore: {}\n\
            Name: {}\n\
            Seed: {:?}",
            self.path.as_os_str().to_str().unwrap(), 
            self.name, self.seed)
    }
}