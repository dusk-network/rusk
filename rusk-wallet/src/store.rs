// This Source Code Form is subject to the terms of the Mozilla Public
// License, v. 2.0. If a copy of the MPL was not distributed with this
// file, You can obtain one at http://mozilla.org/MPL/2.0/.
//
// Copyright (c) DUSK NETWORK. All rights reserved.

use wallet_core::Seed;

use crate::clients::State;

/// Provides a valid wallet seed to `wallet_core`
#[derive(Clone)]
pub(crate) struct LocalStore {
    seed: Seed,
}

impl LocalStore {
    /// Retrieves the seed used to derive keys.
    pub fn get_seed(&self) -> &Seed {
        &self.seed
    }

    pub fn inner_mut(&mut self) -> &mut Seed {
        &mut self.seed
    }
}

impl From<Seed> for LocalStore {
    fn from(seed: Seed) -> Self {
        LocalStore { seed }
    }
}

impl State {
    /// Retrieves the seed used to derive keys.
    pub fn get_seed(&self) -> &Seed {
        self.store().get_seed()
    }
}
