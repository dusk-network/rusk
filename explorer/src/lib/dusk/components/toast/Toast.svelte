<script>
  import { flip } from "svelte/animate";
  import { fly } from "svelte/transition";
  import { Icon } from "$lib/dusk/components";
  import { makeClassName } from "$lib/dusk/string";
  import { toastList, toastTimer } from "./store";
  import { onMount } from "svelte";

  import "./Toast.css";

  /** @type {String | Undefined} */
  export let className = undefined;

  /** @type {Number} */
  export let timer = 2000;

  /** @type {Number} */
  export let flyDuration = 500;

  const classes = makeClassName(["dusk-toast", className]);

  onMount(() => {
    $toastTimer = timer;
  });
</script>

<ul {...$$restProps} class={classes}>
  {#each $toastList as { id, icon, message, type } (id)}
    <li
      in:fly|global={{ duration: flyDuration, x: 200 }}
      out:fly|global={{ duration: flyDuration, x: 200 }}
      animate:flip={{ duration: 200 }}
      class="dusk-toast__item"
    >
      {#if icon}
        <span
          class={`dusk-toast__item-icon-wrapper dusk-toast__item-icon-wrapper--${type}`}
        >
          <Icon className="dusk-toast__item-icon" path={icon} size="default" />
        </span>
      {/if}
      <span class="dusk-toast__item-message">
        {message}
      </span>
    </li>
  {/each}
</ul>
