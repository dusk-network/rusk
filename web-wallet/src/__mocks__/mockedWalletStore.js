import { ProfileGenerator } from "$lib/vendor/w3sper.js/src/profile";

import { stakeInfo } from "$lib/mock-data";

import { mockReadableStore } from "$lib/dusk/test-helpers";

const seed = new Uint8Array(64);
const seeder = () => seed;
const profileGenerator = new ProfileGenerator(seeder);
const profiles = [
  await profileGenerator.default,
  await profileGenerator.next(),
  await profileGenerator.next(),
];
const currentProfile = profiles[0];
const shieldedBalance = {
  spendable: 50_000_000_000_000n,
  value: 2_345_000_000_000n,
};
const publicBalance = { nonce: 1234n, value: shieldedBalance.value / 2n };

/** @type {WalletStoreContent} */
const content = {
  balance: { publicBalance, shieldedBalance },
  currentProfile,
  initialized: true,
  minimumStake: 1_000_000_000_000n,
  profiles,
  stakeInfo,
  syncStatus: {
    error: null,
    from: 0n,
    isInProgress: false,
    last: 0n,
    progress: 0,
  },
};

const mockedWalletStore = mockReadableStore(content);

export default mockedWalletStore;
