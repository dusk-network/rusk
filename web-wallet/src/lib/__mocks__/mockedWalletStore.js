import { ProfileGenerator } from "$lib/vendor/w3sper.js/src/profile";

import { stakeInfo } from "$lib/mock-data";

import { mockReadableStore } from "$lib/dusk/test-helpers";

const seed = new Uint8Array(64);
const seeder = () => seed;
const profileGenerator = new ProfileGenerator(seeder);
const profiles = [
  await profileGenerator.default,
  await profileGenerator.next(),
  await profileGenerator.next(),
];

const currentProfile = profiles[0];
const shielded = { spendable: 50_000_000_000_000n, value: 2_345_000_000_000n };
const unshielded = { nonce: 1234n, value: shielded.value / 2n };

/** @type {WalletStoreContent & { currentProfile: NonNullable<WalletStoreContent["currentProfile"]> }} */
const content = {
  balance: { shielded, unshielded },
  currentProfile,
  initialized: true,
  minimumStake: 1_000_000_000_000n,
  profiles,
  stakeInfo,
  syncStatus: {
    error: null,
    from: 0n,
    isInProgress: false,
    last: 0n,
    progress: 0,
  },
};

const mockedWalletStore = mockReadableStore(content);

export default mockedWalletStore;
