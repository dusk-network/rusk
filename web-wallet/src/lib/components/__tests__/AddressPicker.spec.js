import { afterEach, beforeEach, describe, expect, it, vi } from "vitest";
import { cleanup, fireEvent, render } from "@testing-library/svelte";
import { get } from "svelte/store";

import mockedWalletStore from "$lib/__mocks__/mockedWalletStore";

import { AddressPicker } from "..";

// Mock the toast module
vi.mock("$lib/dusk/components/Toast/store", () => ({
  toast: vi.fn(),
}));

import { toast } from "$lib/dusk/components/Toast/store";

describe("AddressPicker", () => {
  const { currentProfile, profiles } = get(mockedWalletStore);

  const props = { currentProfile, profiles };

  beforeEach(() => {
    Object.assign(navigator, {
      clipboard: {
        writeText: vi.fn().mockResolvedValue(undefined),
      },
    });
    vi.clearAllMocks();
  });

  afterEach(cleanup);

  it("renders the AddressPicker component", () => {
    const { container } = render(AddressPicker, props);

    expect(container.firstElementChild).toMatchSnapshot();
  });

  it("should be able to render the component if the current profile is `null`", () => {
    const { container } = render(AddressPicker, {
      ...props,
      currentProfile: null,
    });

    expect(container.firstElementChild).toMatchSnapshot();
  });

  describe("Multiple Profiles Functionality", () => {
    it("should show dropdown when enableMultipleProfiles is true and trigger is clicked", async () => {
      const { container } = render(AddressPicker, props);

      const trigger = container.querySelector(".address-picker__trigger");
      expect(trigger).toBeTruthy();
      expect(trigger).toHaveAttribute("role", "button");
      expect(trigger).toHaveAttribute("tabindex", "0");

      // Should have chevron icon
      expect(container.querySelector(".address-picker__chevron")).toBeTruthy();

      // Initially dropdown should be closed
      expect(container.querySelector(".address-picker__drop-down")).toBeNull();

      // Click to open dropdown
      if (trigger) {
        await fireEvent.click(trigger);
      }

      // Dropdown should now be visible
      expect(
        container.querySelector(".address-picker__drop-down")
      ).toBeTruthy();
    });

    it("should dispatch setCurrentProfile event when a profile is selected", async () => {
      const { container, component } = render(AddressPicker, props);

      const mockHandler = vi.fn();
      component.$on("setCurrentProfile", mockHandler);

      // Open dropdown
      const trigger = container.querySelector(".address-picker__trigger");
      expect(trigger).toBeTruthy();
      if (trigger) {
        await fireEvent.click(trigger);
      }

      // Click on a profile option
      const profileOptions = container.querySelectorAll(
        ".address-picker__profile-button"
      );
      expect(profileOptions.length).toBeGreaterThan(0);

      if (profileOptions.length > 0) {
        await fireEvent.click(profileOptions[0]);
      }

      expect(mockHandler).toHaveBeenCalledTimes(1);
      expect(mockHandler).toHaveBeenCalledWith(
        expect.objectContaining({
          detail: expect.objectContaining({
            profile: expect.any(Object),
          }),
        })
      );
    });

    it("should close dropdown after profile selection", async () => {
      const { container } = render(AddressPicker, props);

      // Open dropdown
      const trigger = container.querySelector(".address-picker__trigger");
      expect(trigger).toBeTruthy();
      if (trigger) {
        await fireEvent.click(trigger);
      }

      // Verify dropdown is open
      expect(
        container.querySelector(".address-picker__drop-down")
      ).toBeTruthy();

      // Click on a profile option
      const profileOptions = container.querySelectorAll(
        ".address-picker__profile-button"
      );
      if (profileOptions.length > 0) {
        await fireEvent.click(profileOptions[0]);
      }

      // Dropdown should be closed
      expect(container.querySelector(".address-picker__drop-down")).toBeNull();
    });

    it("should handle keyboard navigation", async () => {
      const { container } = render(AddressPicker, props);

      const trigger = container.querySelector(".address-picker__trigger");
      expect(trigger).toBeTruthy();

      // Test Enter key to open dropdown
      if (trigger) {
        await fireEvent.keyDown(trigger, { key: "Enter" });
      }
      expect(
        container.querySelector(".address-picker__drop-down")
      ).toBeTruthy();

      // Test Escape key to close dropdown
      if (trigger) {
        await fireEvent.keyDown(trigger, { key: "Escape" });
      }
      expect(container.querySelector(".address-picker__drop-down")).toBeNull();

      // Test Space key to open dropdown
      if (trigger) {
        await fireEvent.keyDown(trigger, { key: " " });
      }
      expect(
        container.querySelector(".address-picker__drop-down")
      ).toBeTruthy();

      // Test ArrowDown key when closed
      if (trigger) {
        await fireEvent.keyDown(trigger, { key: "Escape" }); // Close first
        await fireEvent.keyDown(trigger, { key: "ArrowDown" });
      }
      expect(
        container.querySelector(".address-picker__drop-down")
      ).toBeTruthy();
    });

    it("should show all profiles in the dropdown list", async () => {
      // Use the first two profiles from the existing profiles array
      const testProfiles = profiles.slice(0, 2);

      const { container } = render(AddressPicker, {
        ...props,
        currentProfile: testProfiles[0],
        profiles: testProfiles,
      });

      // Open dropdown
      const trigger = container.querySelector(".address-picker__trigger");
      expect(trigger).toBeTruthy();
      if (trigger) {
        await fireEvent.click(trigger);
      }

      // Check that all profiles are listed
      const profileOptions = container.querySelectorAll(
        ".address-picker__profile-button"
      );
      expect(profileOptions).toHaveLength(testProfiles.length);

      // Check that current profile is marked as selected
      const selectedProfile = container.querySelector(
        ".address-picker__profile--selected"
      );
      expect(selectedProfile).toBeTruthy();
    });

    it("should handle edge case with single profile", async () => {
      const singleProfile = [profiles[0]];

      const { container } = render(AddressPicker, {
        ...props,
        currentProfile: singleProfile[0],
        profiles: singleProfile,
      });

      // Open dropdown
      const trigger = container.querySelector(".address-picker__trigger");
      expect(trigger).toBeTruthy();
      if (trigger) {
        await fireEvent.click(trigger);
      }

      // Should show one profile option
      const profileOptions = container.querySelectorAll(
        ".address-picker__profile-button"
      );
      expect(profileOptions).toHaveLength(1);
    });
  });

  describe("Copy Address Functionality", () => {
    beforeEach(() => {
      // Reset clipboard mock for each test
      Object.assign(navigator, {
        clipboard: {
          writeText: vi.fn().mockResolvedValue(undefined),
        },
      });
    });

    it("should show copy buttons for each profile's public and shielded addresses", async () => {
      const { container } = render(AddressPicker, props);

      // Open dropdown
      const trigger = container.querySelector(".address-picker__trigger");
      expect(trigger).toBeTruthy();
      if (trigger) {
        await fireEvent.click(trigger);
      }

      // Should have copy button containers for each profile
      const copyButtonContainers = container.querySelectorAll(
        ".address-picker__copy-buttons"
      );
      expect(copyButtonContainers).toHaveLength(profiles.length);

      // Each container should have 2 copy buttons (public and shielded)
      copyButtonContainers.forEach((buttonContainer) => {
        const copyButtons = buttonContainer.querySelectorAll(
          ".address-picker__copy-button"
        );
        expect(copyButtons).toHaveLength(2);
      });
    });

    it("should copy public address when first copy button is clicked", async () => {
      const { container } = render(AddressPicker, props);

      // Open dropdown
      const trigger = container.querySelector(".address-picker__trigger");
      expect(trigger).toBeTruthy();
      if (trigger) {
        await fireEvent.click(trigger);
      }

      // Get the first profile's first copy button (public address)
      const firstCopyButton = container.querySelector(
        ".address-picker__copy-button"
      );
      expect(firstCopyButton).toBeTruthy();

      if (firstCopyButton) {
        await fireEvent.click(firstCopyButton);
      }

      expect(navigator.clipboard.writeText).toHaveBeenCalledTimes(1);
      expect(navigator.clipboard.writeText).toHaveBeenCalledWith(
        profiles[0].account.toString()
      );
    });

    it("should copy shielded address when second copy button is clicked", async () => {
      const { container } = render(AddressPicker, props);

      // Open dropdown
      const trigger = container.querySelector(".address-picker__trigger");
      expect(trigger).toBeTruthy();
      if (trigger) {
        await fireEvent.click(trigger);
      }

      // Get the first profile's second copy button (shielded address)
      const copyButtons = container.querySelectorAll(
        ".address-picker__copy-button"
      );
      expect(copyButtons.length).toBeGreaterThan(1);

      const secondCopyButton = copyButtons[1];
      if (secondCopyButton) {
        await fireEvent.click(secondCopyButton);
      }

      expect(navigator.clipboard.writeText).toHaveBeenCalledTimes(1);
      expect(navigator.clipboard.writeText).toHaveBeenCalledWith(
        profiles[0].address.toString()
      );
    });

    it("should handle clipboard errors gracefully", async () => {
      const mockToast = vi.mocked(toast);
      const clipboardError = new Error("Clipboard access denied");
      clipboardError.name = "NotAllowedError";

      Object.assign(navigator, {
        clipboard: {
          writeText: vi.fn().mockRejectedValue(clipboardError),
        },
      });

      const { container } = render(AddressPicker, props);

      // Open dropdown
      const trigger = container.querySelector(".address-picker__trigger");
      expect(trigger).toBeTruthy();
      if (trigger) {
        await fireEvent.click(trigger);
      }

      // Click first copy button
      const firstCopyButton = container.querySelector(
        ".address-picker__copy-button"
      );
      expect(firstCopyButton).toBeTruthy();

      if (firstCopyButton) {
        await fireEvent.click(firstCopyButton);
      }

      expect(navigator.clipboard.writeText).toHaveBeenCalledTimes(1);
      expect(mockToast).toHaveBeenCalledWith(
        "error",
        "Clipboard access denied",
        expect.any(String)
      );
    });

    it("should show success toast when copy is successful", async () => {
      const mockToast = vi.mocked(toast);

      const { container } = render(AddressPicker, props);

      // Open dropdown
      const trigger = container.querySelector(".address-picker__trigger");
      expect(trigger).toBeTruthy();
      if (trigger) {
        await fireEvent.click(trigger);
      }

      // Click first copy button
      const firstCopyButton = container.querySelector(
        ".address-picker__copy-button"
      );
      expect(firstCopyButton).toBeTruthy();

      if (firstCopyButton) {
        await fireEvent.click(firstCopyButton);
      }

      expect(navigator.clipboard.writeText).toHaveBeenCalledTimes(1);
      expect(mockToast).toHaveBeenCalledWith(
        "success",
        "Address copied",
        expect.any(String)
      );
    });
  });

  describe("Profile Display", () => {
    it("should display profile header with correct format", async () => {
      const { container } = render(AddressPicker, props);

      // Open dropdown
      const trigger = container.querySelector(".address-picker__trigger");
      expect(trigger).toBeTruthy();
      if (trigger) {
        await fireEvent.click(trigger);
      }

      // Check profile headers
      const profileHeaders = container.querySelectorAll(
        ".address-picker__profile-header"
      );
      expect(profileHeaders).toHaveLength(profiles.length);

      // First profile should show "Profile 1"
      expect(profileHeaders[0]).toHaveTextContent("Profile 1");

      // Check if current indicator is shown for current profile
      if (currentProfile === profiles[0]) {
        expect(profileHeaders[0]).toHaveTextContent("(Current)");
      }
    });

    it("should display both public and shielded addresses with proper labels", async () => {
      const { container } = render(AddressPicker, props);

      // Open dropdown
      const trigger = container.querySelector(".address-picker__trigger");
      expect(trigger).toBeTruthy();
      if (trigger) {
        await fireEvent.click(trigger);
      }

      // Check address labels
      const publicLabels = container.querySelectorAll(
        ".address-picker__address-label"
      );

      // Should have both Public and Shielded labels for each profile
      const publicLabelTexts = Array.from(publicLabels).map(
        (label) => label.textContent
      );
      expect(publicLabelTexts).toContain("Public Account");
      expect(publicLabelTexts).toContain("Shielded Account");
    });

    it("should show current profile with proper display format", () => {
      const { container } = render(AddressPicker, props);

      const currentProfileDisplay = container.querySelector(
        ".address-picker__current-profile"
      );
      expect(currentProfileDisplay).toBeTruthy();

      // Should show "Profile X (Default)" format for first profile or "Profile X" for others
      const profileIndex = profiles.indexOf(currentProfile) + 1;
      const expectedText =
        profileIndex === 1
          ? `Profile ${profileIndex} (Default)`
          : `Profile ${profileIndex}`;

      expect(currentProfileDisplay).toHaveTextContent(expectedText);
    });

    it("should handle null currentProfile gracefully", () => {
      const { container } = render(AddressPicker, {
        ...props,
        currentProfile: null,
      });

      const currentProfileDisplay = container.querySelector(
        ".address-picker__current-profile"
      );
      expect(currentProfileDisplay).toBeTruthy();
      expect(currentProfileDisplay).toHaveTextContent("No profile selected");
    });
  });
});
