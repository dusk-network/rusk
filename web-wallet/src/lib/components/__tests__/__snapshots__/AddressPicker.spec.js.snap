// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AddressPicker > renders the AddressPicker component 1`] = `
<div
  class="address-picker"
>
  <div
    aria-expanded="false"
    aria-haspopup="menu"
    aria-label="Select address profile"
    class="address-picker__trigger"
    role="button"
    tabindex="0"
  >
    <span
      class="address-picker__current-profile"
    >
      Profile 1 (Default)
    </span>
     
    <svg
      class="dusk-icon dusk-icon--size--default address-picker__chevron"
      role="graphics-symbol"
      viewBox="0 0 24 24"
    >
      <path
        d="M7.41,8.58L12,13.17L16.59,8.58L18,10L12,16L6,10L7.41,8.58Z"
      />
    </svg>
    
  </div>
   
</div>
`;

exports[`AddressPicker > should be able to render the component if the current profile is \`null\` 1`] = `
<div
  class="address-picker"
>
  <div
    aria-expanded="false"
    aria-haspopup="menu"
    aria-label="Select address profile"
    class="address-picker__trigger"
    role="button"
    tabindex="0"
  >
    <span
      class="address-picker__current-profile"
    >
      No profile selected
    </span>
     
    <svg
      class="dusk-icon dusk-icon--size--default address-picker__chevron"
      role="graphics-symbol"
      viewBox="0 0 24 24"
    >
      <path
        d="M7.41,8.58L12,13.17L16.59,8.58L18,10L12,16L6,10L7.41,8.58Z"
      />
    </svg>
    
  </div>
   
</div>
`;
